src/Facade/Common/IAsyncFacadeService.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Linn.Stores2.Facade.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq.Expressions;
+    using System.Threading.Tasks;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Resources;
+    using PagedList.Core;
+
+    public interface IAsyncFacadeService<T, TKey, TResource, TUpdateResource, in TSearchResource>
+    {
+        Task<IResult<IEnumerable<TResource>>> FilterBy(
+            TSearchResource searchResource, 
+            IEnumerable<string> privileges = null);
+
+        Task<IResult<IEnumerable<TResource>>> FilterBy(
+            TSearchResource searchResource,
+            int numberToTake, 
+            IEnumerable<string> privileges = null);
+
+        Task<IResult<TResource>> FindBy(
+            TSearchResource searchResource, 
+            IEnumerable<string> privileges = null);
+
+        Task<IResult<TResource>> GetById(
+            TKey id, 
+            IEnumerable<string> privileges = null);
+
+        Task<IResult<IEnumerable<TResource>>> GetAll(
+            IEnumerable<string> privileges = null);
+
+        Task<IResult<IEnumerable<TResource>>> Search(
+            string searchTerm, 
+            IEnumerable<string> privileges = null);
+
+        Task<IResult<IPagedList<TResource>>> GetAll(
+            int pageNumber, 
+            int pageSize, 
+            IEnumerable<string> privileges = null);
+
+        Task<IResult<IPagedList<TResource>>> GetAll<TKeySort>(
+            int pageNumber,
```

Comment:
do you think we need all the paged ones in the main service. maybe they're a candidate for a subclass unless other people use them a lot?

Reply:
No I should definitely remove them - I haven't tried any of them yet so I don't even know if they will work with this async repository business!